### ISSUE TITLE
kubejwt authenticator trust domain is not dynamically updated

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#28756]
https://github.com/istio/istio/blob/e57a9b11e2ccf39a581421749f11006c0707dba6/security/pkg/server/ca/authenticate/kube_jwt.go#L62

This causes restart of istiod required here https://preliminary.istio.io/latest/docs/tasks/security/authorization/authz-td-migration/#migrate-trust-domain-without-trust-domain-aliases

**Comment by incfly** [comment]
I see. So we basically need to replace the field value with something like a callback, which provides the hook to access dynamically updated mesh config, when the mounted file for MeshConfig has been updated. Right?

I think if we replace that with something like `MeshHolder` as we did in https://github.com/istio/istio/pull/26185/files#diff-4ab8bb2188afa1dc914f11b5eaab7e1a544454457d25ef3841fdb6f2c23b1d7bR393

That should be sufficient.

---

@chethan-89 would be able/interested to take this one?  Trust domain is a config in the meshconfig. Some old code path like this John linked make it as field only configured at constructor time. Instead, we later find some cases where the mesh config needs to be updated dynamically. Changing the field from a value to a callback which returns the value should fix the issue.
